### PR TITLE
Python3: Use http.client.HTTPMessage.raw_items() to get header fields

### PR DESCRIPTION
**Pull request by ziransun**
This probably is more direct than using .as_string().

### CODE REVIEW

File: xhr/resources/inspect-headers.py [hunk 1]
```diff
@@ -8,13 +8,14 @@ def get_response(raw_headers, filter_value, filter_name):
...
-    # in Python 3.
+    # contained in the header. In Python 3, raw_headers.items() returns a list of 2-tuples
+    # containing all the message’s field headers and values. Here is to construct an equivalent
+    # "headers" variable to support tests in Python 3.
     if PY3:
-        header_list = [
-            isomorphic_encode(s + u'\r\n') for s in raw_headers.as_string().splitlines() if s
-        ]
+        header_list = []
+        for field in raw_headers.items():
```
Comment (annevk) [inline xhr/resources/inspect-headers.py:16]: What does `items()` do though? How the lines are split is somewhat significant.

Comment (Hexcles) [reply xhr/resources/inspect-headers.py:16]: `raw_items` is probably more direct, without additional parsing of header values (which `items` does). Unfortunately it's not documented as it's an "internal" but "public" API.

https://github.com/python/cpython/blob/8021875bbcf7385e651def51bc597472a569042c/Lib/email/message.py#L486

Either way, we don't have much control over how lines and key/value pairs are split. The http stdlib of Python 3 exposes even less raw info than that of Python 2, which is making the case of moving away from stdlib stronger, sigh...

Comment (Hexcles) [reply xhr/resources/inspect-headers.py:16]:
```suggestion
        for field in raw_headers.raw_items():
```

File: xhr/resources/inspect-headers.py [hunk 2]
```diff
@@ -8,13 +8,14 @@ def get_response(raw_headers, filter_value, filter_name):
...
+            header = isomorphic_encode(field[0]) + b": " + isomorphic_encode(field[1]) + b"\r\n"
```
Comment (Hexcles) [inline xhr/resources/inspect-headers.py:17]:
```suggestion
            header = b"%s: %s\r\n" % (isomorphic_encode(field[0]), isomorphic_encode(field[1]))
```

Comment (ziransun) [reply xhr/resources/inspect-headers.py]: Code updated. Thank you!

**Review by Hexcles — APPROVED** [review]
I believe this is better (more "raw" / direct), so LGTM.

But I agree that this is still not perfect as we still don't have full control over the parsing (Python 3 at the very least splits by lines *and* colons for us, which is worse than Python 2 which only splits lines).

**Pull request merged**